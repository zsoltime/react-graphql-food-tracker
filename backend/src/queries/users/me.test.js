'use strict';

const me = require('./me');

describe('me query', () => {
  it('returns the user data if signed in', async () => {
    const fakeUser = {
      id: 'testId',
      email: 'test@email.io',
      permissions: ['USER'],
      profile: null,
    };
    const mockedContext = {
      db: {
        query: { user: jest.fn().mockImplementation(() => fakeUser) },
      },
      request: { userId: 'testUserId' },
    };

    const res = await me(null, null, mockedContext, 'testInfoObject');

    expect(mockedContext.db.query.user).toHaveBeenCalled();
    expect(
      mockedContext.db.query.user.mock.calls[0]
    ).toMatchSnapshot();
    expect(res).toMatchSnapshot();
  });

  it('returns null if user is not signed in', () => {
    const mockedContext = {
      request: {},
    };

    expect(me(null, null, mockedContext)).toBe(null);
  });
});
