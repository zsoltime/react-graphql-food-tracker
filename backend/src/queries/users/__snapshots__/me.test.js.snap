// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`me query returns the user data if signed in 1`] = `
Array [
  Object {
    "where": Object {
      "id": "testUserId",
    },
  },
  "testInfoObject",
]
`;

exports[`me query returns the user data if signed in 2`] = `
Object {
  "email": "test@email.io",
  "id": "testId",
  "permissions": Array [
    "USER",
  ],
  "profile": null,
}
`;
